File: Interviewbit/Tree/cousins-in-binary-tree.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {
    public ArrayList<Integer> solve(TreeNode A, int B) {
        if(A==null) return new ArrayList<>();
        Queue<TreeNode> q=new LinkedList<>();
        q.add(A);
        TreeNode tmp;
        boolean found=false; 
        
        while(!q.isEmpty()&&found==false){
            int size=q.size();
            for(int i=0;i<size;i++){
                tmp=q.poll();
                if(tmp.left!=null &&tmp.left.val==B|| tmp.right!=null && tmp.right.val==B){
                    found=true;
                }else{
                    if(tmp.left) q.add(tmp.left);
                    if(tmp.right) q.add(tmp.right);
                }
            }
        }
        
        ArrayList<Integer> ans=new ArrayList<>();
        for(TreeNode it:q){
            ans.add();
        }
        return ans;
    }
}

File: Interviewbit/Tree/cousins-in-binary-tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
vector<int> Solution::solve(TreeNode* A, int B) {
    if(A==NULL) return {};
    queue<TreeNode* >q;
    q.push(A);
    TreeNode* tmp;
    bool found=false;
    
    while(!q.empty()&&found==false){
        int size=q.size();
        
        for(int i=0;i<size;i++){
            tmp=q.front();
            q.pop();
            if(tmp->left&&tmp->left->val==B ||tmp->right&&tmp->right->val==B){
                found=true;
            }else{
                if(tmp->left) q.push(tmp->left);
                if(tmp->right) q.push(tmp->right);
            }
        }
    }
    vector<int> ans;
    int size=q.size();
    for(int i=0;i<size;i++){
        ans.push_back(q.front()->val);
        q.pop();
    }
    return ans;
}
